domain/src/main/kotlin/no/nav/su/se/bakover/domain/behandling/Behandlingsinformasjon.kt

```diff
@@ -0,0 +1,337 @@
+            }
+
+        override fun isComplete(): Boolean = true
+
+        override fun toDto(): Dto = Dto(status, begrunnelse)
+
+        class Dto(
+            val status: Status,
+            val begrunnelse: String?
+        )
+    }
+
+    data class Formue(
+        val status: Status,
+        val verdiIkkePrimærbolig: Int?,
+        val verdiKjøretøy: Int?,
+        val innskudd: Int?,
+        val verdipapir: Int?,
+        val pengerSkyldt: Int?,
+        val kontanter: Int?,
+        val depositumskonto: Int?
+    ) : Base(), DtoConvertable<Formue.Dto> {
+        enum class Status {
+            Ok,
+            MåInnhenteMerInformasjon
+        }
+
+        override fun isValid(): Boolean =
+            when (status) {
+                Status.Ok ->
+                    verdiIkkePrimærbolig != null &&
+                        verdiKjøretøy !== null &&
+                        innskudd !== null &&
+                        verdipapir !== null &&
+                        pengerSkyldt !== null &&
+                        kontanter !== null &&
+                        depositumskonto !== null
+                Status.MåInnhenteMerInformasjon -> true
+            }
+
+        override fun isComplete(): Boolean = status != Status.MåInnhenteMerInformasjon
+
+        override fun toDto(): Dto = Dto(
+            status,
+            verdiIkkePrimærbolig,
+            verdiKjøretøy,
+            innskudd,
+            verdipapir,
+            pengerSkyldt,
+            kontanter,
+            depositumskonto
+        )
+
+        data class Dto(
+            val status: Status,
+            val verdiIkkePrimærbolig: Int?,
+            val verdiKjøretøy: Int?,
+            val innskudd: Int?,
+            val verdipapir: Int?,
+            val pengerSkyldt: Int?,
+            val kontanter: Int?,
+            val depositumskonto: Int?
+        )
+    }
+
+    data class PersonligOppmøte(
+        val status: Status,
+        val begrunnelse: String?
+    ) : Base(), DtoConvertable<PersonligOppmøte.Dto> {
+        enum class Status {
+            MøttPersonlig,
+            Verge,
+            FullmektigMedLegeattest,
+            FullmektigUtenLegeattest,
+            IkkeMøttOpp
+        }
+
+        override fun isValid(): Boolean = true
+        override fun isComplete(): Boolean = true
+
+        override fun toDto(): Dto = Dto(status, begrunnelse)
+
+        data class Dto(
+            val status: Status,
+            val begrunnelse: String?
+        )
+    }
+
+    data class Sats(
+        val delerBolig: Boolean,
+        val delerBoligMed: Boforhold.DelerBoligMed?,
+        val ektemakeEllerSamboerUnder67År: Boolean?,
+        val ektemakeEllerSamboerUførFlyktning: Boolean?,
+        val begrunnelse: String?
+    ) : Base(), DtoConvertable<Sats.Dto> {
+        fun utledSats() =
+            no.nav.su.se.bakover.domain.beregning.Sats.HØY
+
+        override fun isValid(): Boolean =
```

Comment:
Betyr dette erVilkårOppfylt?

Reply:
Nei, tanken med `isValid`-funksjonene er at de skal si om tilstanden er gyldig. Typ dersom man har valgt at vilkåret ikke er oppfylt så er man nødt til å ha med begrunnelse. Når det er sagt så er jeg ikke sikker på om de reglene jeg har lagt inn her er korrekte.

Reply:
Det er godt mulig vi uansett i første omgang må ha begrunnelse på både ja og nei, siden attestant sannsynligvis må bruke begrunnelsen for å attestere. Men det kan jo Agnethe og de se over når vi får det ut.